controlplane/api/v1beta1/rke2controlplane_types.go

```diff
@@ -436,3 +436,21 @@ func (r *RKE2ControlPlane) GetConditions() clusterv1.Conditions {
 func (r *RKE2ControlPlane) SetConditions(conditions clusterv1.Conditions) {
 	r.Status.Conditions = conditions
 }
+
+// GetDesiredVersion returns the desired version of the RKE2ControlPlane using Spec.Version field as a default field.
```

Comment:
I added a separate function to the RKE2 CP package in case we want to reuse it somewhere else like day 2 operations

Reply:
Would it be valid if `GetDesiredVersion` (cp) and `getDesiredVersion` (bootstrap) return different versions? If this case is possible.